src/ToEnum.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter;
+
+use Laminas\Filter\Exception\RuntimeException;
+use Laminas\Stdlib\ArrayUtils;
+use Traversable;
+use UnitEnum;
+
+use function is_array;
+use function is_int;
+use function is_string;
+use function is_subclass_of;
+
+/**
+ * @psalm-type Options = array{
+ *     enum: class-string<UnitEnum>,
+ * }
+ */
+final class ToEnum implements FilterInterface
+{
+    /**
+     * @var class-string<UnitEnum>|null
+     */
+    private ?string $enumClass = null;
+
+    /**
+     * @param class-string<UnitEnum>|Traversable|Options $enumOrOptions
+     */
+    public function __construct($enumOrOptions)
+    {
+        if ($enumOrOptions instanceof Traversable) {
+            $enumOrOptions = ArrayUtils::iteratorToArray($enumOrOptions);
+        }
+
+        if (
+            is_array($enumOrOptions) &&
+            isset($enumOrOptions['enum'])
+        ) {
+            $this->setEnum($enumOrOptions['enum']);
+
+            return;
+        }
+
+        if (is_string($enumOrOptions)) {
+            $this->setEnum($enumOrOptions);
+        }
+    }
+
+    /**
+     * @param class-string<UnitEnum> $enum
+     */
+    protected function setEnum(string $enum): self
+    {
+        $this->enumClass = $enum;
+
+        return $this;
+    }
+
+    /**
+     * Defined by Laminas\Filter\FilterInterface
+     *
+     * Returns an enum representation of $value or null
+     *
+     * @param  mixed $value
+     * @return UnitEnum|null
```

Comment:
The filters in laminas-filter are typically _normalizers_, and the idea is that you can create a filter chain that will transform a value iteratively via multiple rules. As such, most filters will do the following:

- If they can do some normalization, they will
- Otherwise, they return the value unchanged
- If there is an error performing the operation, they raise an exception